LoveMachine.KK/KoikatsuButtplugController.cs

```diff
@@ -241,6 +241,37 @@ protected override bool IsIdle(int girlIndex) => !supportedModes.Contains(flags.
             RunStrokerLoop(girlIndex, bone);
     }
 
+    internal class KoikatsuButtplugRotatorController : KoikatsuButtplugController
+    {
+        private static readonly HFlag.EMode[] supportedModes =
+        {
+            HFlag.EMode.houshi, HFlag.EMode.sonyu, HFlag.EMode.houshi3P, HFlag.EMode.sonyu3P,
+            HFlag.EMode.houshi3PMMF, HFlag.EMode.sonyu3PMMF
+        };
+
+        private static readonly List<string> supportedAnimations = new List<string>
+        {
+            "WLoop", "SLoop",
+            // anal
+            "A_WLoop", "A_SLoop", "A_OLoop",
+            // orgasm
+            "OLoop", "A_OLoop",
+        }.Union(orgasmAnimations).ToList();
+
+        protected override bool IsIdle(int girlIndex) => !supportedModes.Contains(flags.mode)
+                    || !supportedAnimations.Contains(flags.nowAnimStateName)
+                    || flags.speed < 1;
+
+        protected override bool IsOrgasming(int girlIndex) =>
+            orgasmAnimations.Contains(flags.nowAnimStateName);

```

Comment:
You could get rid of all of these by inheriting from KoikatsuButtplugStrokerController (or by having both inherit from a common base).